jib-cli/src/test/java/com/google/cloud/tools/jib/cli/jar/JarFilesTest.java

```diff
@@ -146,4 +151,222 @@ public void testToJibContainerBuilder_packagedStandard_basicInfo()
                 .build()
                 .getEntries());
   }
+
+  @Test
+  public void testToJibContainerBuilder_explodedLayeredSpringBoot_basicInfo()
+      throws IOException, URISyntaxException, InvalidImageReferenceException {
+    Path springbootJar = Paths.get(Resources.getResource(SIMPLE_SPRING_BOOT_LAYERED).toURI());
+    Path destDir = temporaryFolder.getRoot().toPath();
+    JibContainerBuilder containerBuilder =
+        JarFiles.toJibContainerBuilder(springbootJar, destDir, ProcessingMode.exploded);
+    ContainerBuildPlan buildPlan = containerBuilder.toContainerBuildPlan();
+
+    assertThat(buildPlan.getBaseImage()).isEqualTo("gcr.io/distroless/java");
+    assertThat(buildPlan.getPlatforms()).isEqualTo(ImmutableSet.of(new Platform("amd64", "linux")));
+    assertThat(buildPlan.getCreationTime()).isEqualTo(Instant.EPOCH);
+    assertThat(buildPlan.getFormat()).isEqualTo(ImageFormat.Docker);
+    assertThat(buildPlan.getEnvironment()).isEmpty();
+    assertThat(buildPlan.getLabels()).isEmpty();
+    assertThat(buildPlan.getVolumes()).isEmpty();
+    assertThat(buildPlan.getExposedPorts()).isEmpty();
+    assertThat(buildPlan.getUser()).isNull();
+    assertThat(buildPlan.getWorkingDirectory()).isNull();
+    assertThat(buildPlan.getEntrypoint())
+        .isEqualTo(
+            ImmutableList.of("java", "-cp", "/app", "org.springframework.boot.loader.JarLauncher"));
+    assertThat(buildPlan.getLayers().size()).isEqualTo(4);
+    assertThatExpectedEntriesPresentInNonSnapshotLayer_SpringBoot(
+        ((FileEntriesLayer) buildPlan.getLayers().get(0)).getEntries(), destDir);
+    assertThatExpectedEntriesPresentInLoaderLayer_SpringBoot(
+        ((FileEntriesLayer) buildPlan.getLayers().get(1)).getEntries(), destDir);
+    assertThatExpectedEntriesPresentInSnapshotLayer_SpringBoot(
+        ((FileEntriesLayer) buildPlan.getLayers().get(2)).getEntries(), destDir);
+    assertThat(((FileEntriesLayer) buildPlan.getLayers().get(3)).getEntries())
+        .containsExactlyElementsIn(
+            FileEntriesLayer.builder()
+                .addEntry(destDir.resolve("META-INF/"), AbsoluteUnixPath.get("/app/META-INF/"))
+                .addEntry(
+                    destDir.resolve("META-INF/MANIFEST.MF"),
+                    AbsoluteUnixPath.get("/app/META-INF/MANIFEST.MF"))
+                .addEntry(destDir.resolve("BOOT-INF/"), AbsoluteUnixPath.get("/app/BOOT-INF/"))
```

Comment:
I also commented this below, but we should try to exclude directories that don't belong to this layer.

Reply:
Good point. Applied this change.